server/src/main/java/org/elasticsearch/transport/NodeDisconnectedException.java

```diff
@@ -34,4 +36,10 @@ public NodeDisconnectedException(StreamInput in) throws IOException {
     public Throwable fillInStackTrace() {
         return this;
     }
+
+    @Override
+    public RestStatus status() {
+        Throwable cause = getCause();
+        return cause != null ? ExceptionsHelper.status(cause) : RestStatus.BAD_GATEWAY;
+    }
```

Comment:
We have a separate task open to make _all_ descendants of `ConnectTransportException` map onto a 502. You need take no action on this front, and definitely shouldn't be doing something like this to specific subclasses.